pages/settings/admin/tasks.vue

```diff
@@ -0,0 +1,248 @@
+
+    rawTasks.forEach((task) => {
+      if (task.IsHidden) {
+        rawTasks.splice(rawTasks.indexOf(task));
+      }
+    });
+
+    rawTasks.sort((a: TaskInfo, b: TaskInfo) => {
+      if (a.Category && b.Category) {
+        if (a.Category < b.Category) {
+          return -1;
+        }
+        if (a.Category > b.Category) {
+          return 1;
+        }
+      }
+      return 0;
+    });
+
+    const allTasks = groupBy(rawTasks, 'Category');
+    const taskStatus = {} as { [k: string]: TaskState };
+    const taskProgresses = {} as { [k: string]: number };
+    rawTasks.forEach((task) => {
+      if (task.Id) {
+        taskStatus[task.Id] = task.State as TaskState;
+        if (task.CurrentProgressPercentage) {
+          taskProgresses[task.Id] = task.CurrentProgressPercentage;
+        } else {
+          taskProgresses[task.Id] = 0;
+        }
+      }
+    });
+
+    return { allTasks, taskStatus, taskProgresses };
+  },
+  data() {
+    const uuid = uuidv4();
+    return {
+      uuid,
+      idle: TaskState.Idle,
+      cancelling: TaskState.Cancelling,
+      running: TaskState.Running,
+      allTasks: {} as GroupedTasks,
+      taskStatus: {} as { [k: string]: TaskState },
+      taskProgresses: {} as { [k: string]: number },
+      loading: false,
+      interval: 0
+    };
+  },
+  head() {
+    return {
+      title: this.$store.state.page.title
+    };
+  },
+  created() {
+    this.setPageTitle({
+      title: this.$t('settingsSections.admin.scheduledTasks.name')
+    });
+  },
+  mounted() {
+    /**
+     * TODO: Use WebSocket for this once it's available in the API.
+     */
+    this.interval = window.setInterval(this.pollUpdates, 700);
+  },
+  destroyed() {
+    window.clearInterval(this.interval);
+  },
+  methods: {
+    ...mapActions('page', ['setPageTitle']),
+    getRuntimeText(task: TaskInfo): string {
+      if (
+        task.LastExecutionResult?.StartTimeUtc &&
+        task.LastExecutionResult?.EndTimeUtc
+      ) {
+        const delta = this.$dateFns.formatDistance(
+          Date.parse(task.LastExecutionResult?.StartTimeUtc),
+          Date.parse(task.LastExecutionResult?.EndTimeUtc),
+          {
+            locale: this.$i18n.locale
+          }
+        );
+        const time = this.$dateFns.formatDistance(
+          Date.parse(task.LastExecutionResult?.StartTimeUtc),
+          Date.now(),
+          {
+            addSuffix: true,
+            locale: this.$i18n.locale
+          }
+        );
+        return this.$i18n.t('settings.tasks.lastRan', { time, delta });
+      } else {
+        return this.$i18n.t('settings.tasks.noRuntimeInfo');
+      }
+    },
+    async pollUpdates(): Promise<void> {
+      const rawTasks = (await this.$api.scheduledTasks.getTasks())
+        .data as TaskInfo[];
+      rawTasks.forEach((task: TaskInfo) => {
+        if (!task.IsHidden && task.Id) {
+          this.taskStatus[task.Id] = task.State as TaskState;
```

Comment:
I feel like this assertion of state is wrong.

I don't think we need TaskState here, if we do the api spec is incorrect?